### PR TITLE
Complete migration to OWNERS file

### PR DESCRIPTION
**Pull request by jmhbnz**
As part of our setup for sig-etcd we intend to adopt the kubernetes project `OWNERS` file approach for managing repository and github org permissions automatically as code.

In https://github.com/etcd-io/website/pull/725 we introduced the `OWNERS` file for etcd website.

This pull request is a tidy-up to remove the old `MAINTAINERS` file approach.

Sub task under: https://github.com/etcd-io/etcd/issues/16367

### CODE REVIEW

**Review by serathius — APPROVED** [review]

**Review by ahrtr — APPROVED** [review]
LGTM

thx @jmhbnz

**Comment by jberkus** [comment]
So, somewhat ironically, I can't approve this because in the current in-between state my approval status is broken.

/lgtm nevertheless

**Comment by ahrtr** [comment]
> So, somewhat ironically, I can't approve this because in the current in-between state my approval status is broken

Somehow you were not a maintainer in the team maintainers-website. Just changed your role to maintainer.  Does it work now?

**Pull request merged**
